### ISSUE TITLE
RSA <= 4.0 supports Python2 - Moto pulls in 4.5

### ISSUE THREAD

**Issue by bblommers** [spulec/moto#3104]
RSA is a transitive dependency pulled in from python-jose. Python-Jose supports Python2 and requires any version of RSA - even though [RSA 4.0 is the last version to support Python2](https://pypi.org/project/rsa/#description).

This causes `make init` to fail on Macs, as observed in #3086.

To make things interesting - our Travis builds for Python2 work just fine, pulling in the latest RSA version, 4.5.

Ideally this is fixed upstream by opening a PR in python-jose - just raising it here for awareness.

**Comment by jweite** [comment]
@bblommers, the 3.2 release of python-jose is now getting picky about ecdsa version:

```
(py3.7) $ make init
ecdsa 0.15 is installed but ecdsa<0.15 is required by {'python-jose'}
```

I personally modified python-jose version in setup.py to be <3.2.0 (and included rsa<=4.0 in the PY2 install_requires to deal with the above).

**Comment by bblommers** [comment]
This was fixed with #3697 and #3263
